rust/noosphere-core/src/stream/block.rs

```diff
@@ -0,0 +1,42 @@
+use anyhow::Result;
+use cid::Cid;
+use libipld_cbor::DagCborCodec;
+use libipld_core::raw::RawCodec;
+use noosphere_storage::BlockStore;
+use tokio_stream::{Stream, StreamExt};
+
+/// Helper to put blocks from a [Stream] into any implementor of [BlockStore]
+///
+/// Implementation note: this is a stand-alone helper because defining this
+/// async function on a trait (necessitating use of `#[async_trait]`) creates an
+/// unergonomic `Send` bound on returned [Future].
```

Comment:
I think I've circumvented this caveat for cross platform streams recently (in storage changes, or body chunk stream work) -- will evaluate during the rebase process to see